smartsim/experiment.py

```diff
@@ -218,15 +223,73 @@ def execute_dispatch(job: Job) -> LaunchedJobID:
                 launch_config = dispatch.create_new_launcher_configuration(
                     for_experiment=self, with_arguments=args
                 )
-            # Save the underlying launcher instance. That way we do not need to
-            # spin up a launcher instance for each individual job, and it makes
-            # it easier to monitor job statuses
+            id_ = launch_config.start(exe, env)
+            # Save the underlying launcher instance and launched job id. That
+            # way we do not need to spin up a launcher instance for each
+            # individual job, and the experiment can monitor job statuses.
             # pylint: disable-next=protected-access
-            self._active_launchers.add(launch_config._adapted_launcher)
-            return launch_config.start(exe, env)
+            self._active_launchers[launch_config._adapted_launcher].add(id_)
+            return id_
 
         return execute_dispatch(job), *map(execute_dispatch, jobs)
 
+    def get_status(self, *ids: LaunchedJobID) -> tuple[SmartSimStatus, ...]:
+        """Get the status of jobs launched through the `Experiment` from their
+        launched job id returned when calling `Experiment.start`.
+
+        The `Experiment` will map the launched ID back to the launcher that
+        started the job and request a status update. The order of the returned
+        statuses exactly matches the order of the launched job ids.
+
+        If the `Experiment` cannot find any launcher that started the job
+        associated with the launched job id, then a
+        `SmartSimStatus.STATUS_NEVER_STARTED` status is returned for that id.
```

Comment:
Agreed offline to split up the `SmartSimStatus` enum into a `SmartSimStatus` enum and an `SmartSimStatusError` (name pending) enum. Launchers will be able to report a `SmartSimStatus`, but any invalid input would return a `SmartSimStatusError` enum.

The reason for this split is because it was requested that this method does not raise!